packages/core/src/Site/index.ts

```diff
@@ -1184,7 +1256,7 @@ class Site {
     context.numPagesGenerated += 1;
 
     if (pageGenerationQueue.length) {
-      pageGenerationQueue.pop()();
+      pageGenerationQueue.pop()!();
```

Comment:
Reason for this change?

Reply:
`pageGenerationQueue.pop()` may be undefined if the array is empty, which TS will complain about.
We check that this isn't the case using the `.length` check and so we can assert that it's not undefined using `!`